extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/MSQTaskQueryMaker.java

```diff
@@ -281,6 +284,13 @@ public QueryResponse<Object[]> runQuery(final DruidQuery druidQuery)
                .tuningConfig(new MSQTuningConfig(maxNumWorkers, maxRowsInMemory, rowsPerSegment, indexSpec))
                .build();
 
+    final SegmentSource segmentSources = MultiStageQueryContext.getSegmentSources(sqlQueryContext);
+    if (MSQControllerTask.isReplaceInputDataSourceTask(querySpec) && SegmentSource.REALTIME.equals(segmentSources)) {
+      throw DruidException.forPersona(DruidException.Persona.USER)
+                          .ofCategory(DruidException.Category.INVALID_INPUT)
+                          .build("REALTIME segment sources cannot be queried while reindexing into the same datasource.");
```

Comment:
Add the data source name to the error message in both places. 
Also can we push up the validation to a common utils class so that it can be used in both places.